app/screens/convert_gm_to_channel/team_selector.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Platform} from 'react-native';
+
+import OptionItem from '@components/option_item';
+import {Screens} from '@constants';
+import {useTheme} from '@context/theme';
+import {dismissBottomSheet, goToScreen} from '@screens/navigation';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        teamSelector: {
+            borderTopWidth: 1,
+            borderBottomWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+    };
+});
+
+type Props = {
+    commonTeams: Team[];
```

Comment:
We use `commonTeams` just for two things:
- Getting the display name of the team
- Calling `onSelectTeam` with the team instead of the id

If feel things would be much simpler if we leave that part of the logic to the parent, and make this component "dumber". This component only needs the selected team display name, and a `onSelectTeam` function that receives a teamId. It should need no more.

But 0/5. If the parent component gets too complicated, it may make sense to have this logic here.

Reply:
We also pass common teams to team selector list screen which is opened from this component, thats why we need to keep this.